azurelinuxagent/pa/deprovision/default.py

```diff
@@ -189,6 +189,11 @@ def reset_hostname(self, warnings, actions):
         actions.append(DeprovisionAction(self.osutil.set_dhcp_hostname, 
                                          localhost))
 
+    def check_ifcfg_hwaddr(self, warnings):
+        for f in glob.glob("/etc/sysconfig/network-scripts/ifcfg-*"):
+            if os.path.isfile(f) and fileutil.findre_in_file(f, '^\s*HWADDR='):
```

Comment:
You should probably do this for MACADDR as well (see [the CentOS docs](https://www.centos.org/docs/5/html/Deployment_Guide-en-US/s1-networkscripts-interfaces.html)).

As I read that doc, the file syntax mandates there be no whitespace before the equal-sign, so this is good.
```
if os.path.isfile(f) and fileutil.findre_in_file(f, '^\s*(HW|MAC)ADDR='):
```